### PR TITLE
Update DFT definition to more closely align with PyTorch by enabling axis attribute, and arbitrary tensor rank.

### PR DESCRIPTION
**Pull request by smk2007**
Update DFT definition to more closely align with PyTorch by enabling axis attribute, and arbitrary tensor rank.

### CODE REVIEW

File: winml/test/api/LearningModelSessionAPITest.cpp [hunk 1]
```diff
@@ -495,19 +499,38 @@ static void DiscreteFourierTransform(bool is_onesided = false) {
...
 
-  // Check results
-  printf("Output.Spectra\n");
-  auto y_tensor = result.Outputs().Lookup(L"Output.Spectra").as<TensorFloat>();
-  auto y_ivv = y_tensor.GetAsVectorView();
-  for (int i = 0; i < output_shape[0] * output_shape[1] * 2; i += 2) {
-    printf("(%f + %fi), ", y_ivv.GetAt(i), y_ivv.GetAt(i + 1));
-  }
-  printf("\n");
+  // // Check results
```
Comment (ryanlai2) [inline winml/test/api/LearningModelSessionAPITest.cpp:522]: How do you check results now?

Before, the test would print to output.

File: onnxruntime/contrib_ops/cpu/signal/dft.h [hunk 2]
```diff
@@ -8,16 +8,20 @@ namespace contrib {
 
 class DFT final : public OpKernel {
   bool is_onesided_ = true;
+  int64_t axis_ = 0;
  public:
   explicit DFT(const OpKernelInfo& info) : OpKernel(info) {
     is_onesided_ = static_cast<bool>(info.GetAttrOrDefault<int64_t>("onesided", 0));
+    axis_ = static_cast<bool>(info.GetAttrOrDefault<int64_t>("axis", 0));
```
Comment (ryanlai2) [inline onnxruntime/contrib_ops/cpu/signal/dft.h:15]: >< [](http://example.com/codeflow?start=23&length=1)

I think the static_cast needs to be removed :)

File: onnxruntime/contrib_ops/cpu/signal/dft.cc [hunk 3]
```diff
@@ -39,20 +39,16 @@ ONNX_OPERATOR_KERNEL_EX(
     kMSExperimentalDomain,
     1,
     kCpuExecutionProvider,
-    KernelDefBuilder().MayInplace(0, 0).TypeConstraint("T", BuildKernelDefConstraints<float, double>()),
+    KernelDefBuilder().MayInplace(0, 0).TypeConstraint("T1", BuildKernelDefConstraints<float, double>())
+                                       .TypeConstraint("T2", BuildKernelDefConstraints<int64_t>()),
     STFT);
 
 static bool is_real_valued_signal(const onnxruntime::TensorShape & shape) {
-  // The first dimention is the batch size
```
Comment (ryanlai2) [inline onnxruntime/contrib_ops/cpu/signal/dft.cc:46]: Nit: I liked the comments from before that explained the dimensions

File: onnxruntime/contrib_ops/cpu/signal/dft.h (hunk 2 above)
Comment (smk2007) [reply onnxruntime/contrib_ops/cpu/signal/dft.h:15]: yes!

File: onnxruntime/contrib_ops/cpu/signal/dft.cc (hunk 3 above)
Comment (smk2007) [reply onnxruntime/contrib_ops/cpu/signal/dft.cc:46]: This should be explained in the operator schema in detail.

**Pull request merged**